qiskit/pulse/compiler/passes/schedule.py

```diff
@@ -0,0 +1,189 @@
+        return passmanager_ir
+
+    def _schedule_recursion(self, prog: SequenceIR) -> None:
+        """Recursively schedule the IR.
+
+        Nested IR objects must be scheduled first, so we traverse the IR,
+        and recursively schedule the IR objects.
+        After all nested IR objects are scheduled, we apply the scheduling strategy to the
+        current object.
+
+        Arguments:
+            prog: The IR object to be scheduled.
+        """
+        for elem in prog.elements():
+            if isinstance(elem, SequenceIR):
+                self._schedule_recursion(elem)
+
+        self._schedule_single_program(prog.alignment, prog.sequence, prog.time_table)
+
+    @singledispatchmethod
+    def _schedule_single_program(
+        self, alignment: AlignmentKind, sequence: PyDAG, time_table: defaultdict
+    ) -> None:
+        """Concretely schedule the IR object.
+
+        The ``time_table`` argument is mutated to include the initial time of each element of
+        ``sequence``, according to the structure of ``sequence`` and the alignment.
+        The function assumes that nested IR objects are already scheduled.
+
+        ``sequence`` is assumed to have the following structure - node 0 marks the beginning of the
+        sequence, while node 1 marks the end of it. All branches of the graph originate from node 0
+        and end at node 1.
+
+        Arguments:
+            alignment: The alignment of the IR object.
+            sequence: The sequence of the IR object.
+            time_table: The time_table of the IR object.
+        """
+        raise NotImplementedError
+
+    # pylint: disable=unused-argument
+    @_schedule_single_program.register(AlignLeft)
+    @_schedule_single_program.register(AlignSequential)
+    def _schedule_recursion_align_left(
+        self, alignment: AlignmentKind, sequence: PyDAG, time_table: defaultdict
+    ) -> None:
+        """Concretely schedule the IR object, aligning to the left.
+
+        The ``time_table`` argument is mutated to include the initial time of each element of
+        ``sequence``, according to the structure of ``sequence`` and aligning to the left.
+        The function assumes that nested IR objects are already scheduled.
+
+        ``sequence`` is assumed to have the following structure - node 0 marks the beginning of the
+        sequence, while node 1 marks the end of it. All branches of the graph originate from node 0
+        and end at node 1.
+
+        Arguments:
+            alignment: The alignment of the IR object.
+            sequence: The sequence of the IR object.
+            time_table: The time_table of the IR object.
+
+        Raises:
+            PulseCompilerError: If the sequence is not sequenced as expected.
+        """
+        nodes = list(topological_sort(sequence))
+        if nodes[0] != 0 or nodes[-1] != 1:
+            raise PulseCompilerError(
+                "The sequence is not sequenced as expected. Use SetSequence pass."
```

Comment:
```suggestion
                "The pulse program is not sequenced as expected. "
                "Insert SetSequence pass in advance of the SchedulePass."
```
The end users might not be aware of _sequence_.